src/t8_geometry/t8_geometry_helpers.c

```diff
@@ -372,6 +372,223 @@ t8_geom_get_triangle_scaling_factor (int edge_index, const double *tree_vertices
   return scaling_factor;
 }
 
+double
+t8_geom_get_scaling_factor_of_edge_on_face_prism (const int edge, const int face, const double *ref_coords)
+{
+  double orthogonal_direction;
+  double max_orthogonal_direction;
+  double scaling_factor;
+  /* Save the orthogonal direction and the maximum of that direction
+   * of an edge in reference space on one of the neighbouring faces. 
+   *           /|
+   *          / |
+   *         /  |
+   *        /   |
+   *       /    |--edge
+   *      /   --|--face
+   *     /      |
+   *    /    <~~| orthogonal direction
+   *   /<----o--| maximum othogonal direction
+   *  /_________|
+   */
+  switch (edge) {
```

Comment:
Instead of the encapsulated switches, you could also create a 
`const double orthogonal_direction[num_edges][num_faces]` and put the values there. 
Then evaluation is just a look up and the code becomes more clean. And you can use an assertion to check if the combination of edges and faces match.